### ISSUE TITLE
[Accessibility] Carousel block - Carousel is not wrapped in a section tag (or role=region)

### ISSUE THREAD

**Issue by andeng1106** [gambitph/Stackable#3248]
**Describe the bug**
- Carousel is not wrapped in a `<section>` tag nor have a `role=region`
- See these links below for more information:
     - https://www.w3.org/WAI/tutorials/carousels/structure/)
     - https://www.smashingmagazine.com/2023/02/guide-building-accessible-carousels/#regions-and-groups
- Currently Stackable Carousel block is wrapped in `<div>` (which works) but doesn't have a `role=region`

**To Reproduce**
1. Add a new page
2. Add a Carousel block
3. Save page and view frontend
4. Inspect element
5. Select the Carousel block to inspect
6. Check if there's a `role=region` present or if Carousel's main tag is `<section>`

**Expected behavior**
- Carousel should be wrapped in a section tag 
OR 
- If div needs to be retained, there should be role="region" present

**Screenshots**

https://github.com/user-attachments/assets/63792897-59ff-4ddd-a7ad-bb8210ddeacc
